src/utils/hex-utils.ts

```diff
@@ -6,10 +6,15 @@ export const toHexString32 = (
   value: string | number | BigNumber | boolean
 ): string => {
   if (typeof value === 'string' && value.startsWith('0x')) {
-    return '0x' + remove0x(value).padStart(64, '0').toLowerCase()
```

Comment:
This function probably needs to be cleaned up in the future but whatever for now.